lib/util/stream_utils.js

```diff
@@ -499,6 +499,33 @@ shaka.util.StreamUtils = class {
   }
 
 
+  /**
+   * Queries mediaCapabilities for the decoding info for that decoding config,
+   * and assigns it to the given variant.
+   * If that query has been done before, instead return a cached result.
+   * @param {!shaka.extern.Variant} variant
+   * @param {!MediaDecodingConfiguration} decodingConfig
+   * @private
+   */
+  static async getDecodingInfosForVariant_(variant, decodingConfig) {
+    try {
+      const cacheKey = JSON.stringify(decodingConfig);
```

Comment:
This is a problematic key.  If the fields are serialized in a different order, this will fail to match.

It's better than no caching, but this needs work.  Please file an issue to follow-up on this.

Reply:
(There's an entire ToTT episode on this subject, BTW, in context of comparing objects in tests using JSON.stringify.  But it still applies here.)